fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -5738,11 +5739,17 @@ public long loadCluster(DataInputStream dis, long checksum) throws IOException,
                 // for adding BE to some Cluster, but loadCluster is after loadBackend.
                 cluster.setBackendIdList(latestBackendIds);
 
-                final InfoSchemaDb db = new InfoSchemaDb(cluster.getName());
-                db.setClusterName(cluster.getName());
-                idToDb.put(db.getId(), db);
-                fullNameToDb.put(db.getFullName(), db);
-                cluster.addDb(db.getFullName(), db.getId());
+                String dbName = ClusterNamespace.getFullName(cluster.getName(), InfoSchemaDb.DATABASE_NAME);
+                if (!fullNameToDb.containsKey(dbName)) {
+                    final InfoSchemaDb db = new InfoSchemaDb(cluster.getName());
+                    db.setClusterName(cluster.getName());
+                    String errMsg = "InfoSchemaDb id shouldn't larger than 10000, please restart your FE server";
+                    // ensure InfoSchemaDb meta don't overwrite the normal db meta
+                    Preconditions.checkState(db.getId() < NEXT_ID_INIT_VALUE, errMsg);
+                    idToDb.put(db.getId(), db);
+                    fullNameToDb.put(db.getFullName(), db);
+                }
+                cluster.addDb(dbName, fullNameToDb.get(dbName).getId());
```

Comment:
I'm not sure why this change can fix database lost, could you please add some comment to explain it?

Reply:
How do you guarantee that the InfoSchemaDb you created here has id less than NEXT_ID_INIT_VALUE?
And I think since you've already changed the `saveDb()` method, which will skip InfoSchemaDb by name, not by ID. Here we do not need to check the ID anymore.

Reply:
Oh, I see. The id of InfoSchemaDb is under another id serial.
So the problem is that if we create more than 10000 clusters, this may still be a problem?

Reply:
> So the problem is that if we create more than 10000 clusters, this may still be a problem? However it is almost impossible to happen.

Yes. I also considered this problem.  I think cluster will be obsolete after https://github.com/apache/incubator-doris/issues/1723, so I don't add a check when create cluster.

If you think the check is necessary, I could add a check.

Reply:
It' OK. Don' t have to change~